helios-services/src/main/java/com/spotify/helios/master/MasterService.java

```diff
@@ -308,13 +316,40 @@ private void logBanner() {
    * @return A zookeeper client.
    */
   private ZooKeeperClient setupZookeeperClient(final MasterConfig config) {
+    ACLProvider aclProvider = null;
+    List<AuthInfo> authorization = null;
+
+    if (config.isZooKeeperEnableAcls()) {
```

Comment:
this block is repeated in AgentService and MasterService, just with a reversal of the labels, maybe it can be moved into the CuratorClientFactory?